src/test/secure-tunnel-tests.ts

```diff
@@ -226,6 +206,10 @@ describe('https tunnel', function() {
     return true;
   });
 
+  function isError(error: any): error is NodeJS.ErrnoException {
```

Comment:
@Thilo-Acrolinx I think the return type should be boolean. Based on mdn docs https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/instanceof

Reply:
@abhijeetnarvekar this function is the best solution I was able to find since there was a compilation error due to a Typescript update. Apparently nowadays the caught error is of type _unknown_ and thus `error.code` cannot be accessed any longer.

This type predicate (https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) was described on this page here: https://dev.to/jdbar/the-problem-with-handling-node-js-errors-in-typescript-and-the-workaround-m64 and therefore afaik the function cannot return boolean, it results in a compilation error.

But I am open for better solutions, this is indeed an ugly workaround.